src/util/pg.py

```diff
@@ -1621,3 +1622,76 @@ def create_id_sequence(cr, table, set_as_default=True):
                 table=table_sql,
             )
         )
+
+
+def bulk_update_table(cr, table, columns, mapping, key_col="id"):
+    """
+    Update table based on mapping.
+
+    Each `mapping` entry defines the new values for the specified `columns` for the row(s)
+    whose `key_col` value matches the key.
+
+    .. example::
+
+       .. code-block:: python
+
+          # single column update
+          util.bulk_update_table(cr, "res_users", "active", {42: False, 27: True})
+
+          # multi-column update
+          util.bulk_update_table(
+              cr,
+              "res_users",
+              ["active", "login"],
+              {
+                  "administrator": [True, "admin"],
+                  "demouser": [True, "demo"],
+              },
+              key_col="login",
```

Comment:
`key_col` is always the key of the mapping... unless I'm missing something.
```suggestion
                  2: [True, "admin"],
                  6: [True, "demo"],
              },
              key_col="id",
```